src/bridge_stream.rs

```diff
@@ -25,66 +28,148 @@ where
     T: Stream<Item = StreamItem<E>> + Sized + Unpin,
 {
     stream: T,
-    current: Vec<Packet>,
+    current: Vec<Vec<Packet>>,
     complete: bool,
 }
 
+impl<E: Fail + Sync + Send, T: Stream<Item = StreamItem<E>> + Sized + Unpin>
+    BridgeStreamState<E, T>
+{
+    fn is_complete(&self) -> bool {
+        self.complete && self.current.is_empty()
+    }
+
+    fn spread(&self) -> Duration {
+        let min = self.current.first().map(|s| s.first()).flatten();
+
+        let max = self.current.first().map(|s| s.first()).flatten();
```

Comment:
Min and max are the same. Should max be from `last`?

Reply:
derp